internal/payloads/error_payload.go

```diff
@@ -66,40 +69,89 @@ func PubkeyDuplicateError(ctx context.Context, message string, err error) *Respo
+	if errors.Is(err, httpClients.MoreThanOneAuthenticationForSourceErr) {
+		return 500, "sources returned more than one authentication"
+	} else if errors.Is(err, httpClients.ApplicationNotFoundErr) {
+		return 404, "sources application not found"
+	} else if errors.Is(err, httpClients.ApplicationTypeNotFoundErr) {
+		return 404, "unexpected source type"
+	} else if errors.Is(err, httpClients.SourceNotFoundErr) {
+		return 404, "source not found"
+	} else if errors.Is(err, httpClients.AuthenticationSourceAssociationErr) {
+		return 500, "authentication associated to source id not found"
+	} else if errors.Is(err, httpClients.AuthenticationForSourcesNotFoundErr) {
+		return 404, "authentication for source not found"
+	}
+	return 0, ""
+}
+
+func ImageBuilderHelper(err error) (int, string) {
+	if errors.Is(err, httpClients.ComposeNotFoundErr) {
+		return 404, "image builder did not find image compose"
+	} else if errors.Is(err, httpClients.ImageStatusErr) {
+		return 500, "image builder has not finished the build of requested image"
+	} else if errors.Is(err, httpClients.UnknownImageTypeErr) {
+		return 500, "unknown image type"
+	} else if errors.Is(err, httpClients.AmiNotFoundInStatusErr) {
+		return 404, "image builder did not find AMI in status"
+	} else if errors.Is(err, httpClients.NameNotFoundInStatusErr) {
+		return 404, "image builder did not find image name in image status"
+	} else if errors.Is(err, httpClients.IDNotFoundInStatusErr) {
+		return 404, "image builder did not find project id in image status"
+	}
+	return 0, ""
+}
+
+func Response(ctx context.Context, status int, message string, err error) *ResponseError {
 	if logger := ctxval.Logger(ctx); logger != nil {
-		logger.Error().Msg(msg)
+		logger.Error().Msg(message)
 	}
 	return &ResponseError{
 		HTTPStatusCode: status,
-		Message:        msg,
+		Message:        message,
 		TraceId:        ctxval.TraceId(ctx),
 		Error:          err,
 	}
 }
 
-func NewClientInitializationError(ctx context.Context, message string, err error) *ResponseError {
-	msg := fmt.Sprintf("HTTP client initialization error: %s: %v", message, err)
-	if logger := ctxval.Logger(ctx); logger != nil {
-		logger.Error().Msg(msg)
+func NewError(ctx context.Context, err error) *ResponseError {
+	if errors.Is(err, clients.UnknownAuthenticationTypeErr) {
+		return Response(ctx, 500, "unknown authentication type", err)
 	}
-	return &ResponseError{
-		HTTPStatusCode: 500,
-		Message:        msg,
-		TraceId:        ctxval.TraceId(ctx),
-		Error:          err,
+	if status, message := ImageBuilderHelper(err); status != 0 {
+		return Response(ctx, status, message, err)
 	}
+	if status, message := SourcesErrorHelper(err); status != 0 {
+		return Response(ctx, status, message, err)
+	}
+	if status, message := ClientErrorHelper(err); status != 0 {
+		return Response(ctx, status, message, err)
+	}
+	return Response(ctx, 500, "HTTP service returned unknown client error", err)
 }
```

Comment:
I see one major issue: this function is called `NewError` thus most of us will shortly start it using for ALL errors while what it actually does is setting up details for HTTP client errors only. I suggest to rename it to `NewClientError` for now and work with it for some weeks and if this approach proves to be valid, we can even extend this to all other components (DAO, generic errors).